core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1003,6 +1022,7 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
    * Returns a successful effect with the head of the list if the list is
    * non-empty or fails with the error `None` if the list is empty.
    */
+  // TODO: Should we change this to work with any Iterable?
   final def head[B](implicit ev: A IsSubtypeOfOutput List[B]): ZIO[R, Option[E], B] =
```

Comment:
I don't know if we want to encourage potentially inefficient data access patterns. The promise of an `Iterable` is really only that you can iterate over it. If this was a frequently used method would be happy to put that aside for ergonomics but don't know if that is really the case.